### DISCUSSION TITLE
Generate paginated content on the new Razor SSG

### DISCUSSION THREAD

**Discussion by nukedbit** [ServiceStack/Discuss#64] [Q&A] [answered]
I'm looking to generate paginated content for my blog section.
I see on the template code that i can easily access `Blog.GetPosts();` but i was wondering how i should handle the pagination template.
There is a way to tell a cshtml to generate paginated static content for example having a Blog.cshtml that end up generated as /page_1, /page_2 etc..
Or the way currently is to have page1.cshtml etc..?
Thanks

**Comment by nukedbit** [comment, marked as answer]
Ok my bad i think i have found out how to do, for some reason dotnet watch wasent picking my changes, and i tought that route parameter werent working, i see you use it on the Page.cshtml example.

Basically i think i should use `@page "/pages/{index:int?}"` and generate the links to move between pages right?

**Reply by Layoric** [reply]
Yup, that is a good way forward. If you have any trouble let us know and I'll try put together a sample.

**Comment by mythz** [comment]
One limitation of Razor Pages is that it can only support a single route, so to handle pagination in https://diffusion.works (Blazor Diffusion rewritten in Razor SSG/Vue) is to have different Razor Pages for the first and another for the paginated page:

E.g https://diffusion.works/latest is handled by [/Pages/Latest.cshtml](https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/Latest.cshtml)

```csharp
﻿@page "/latest"
```

and other paginated pages like https://diffusion.works/latest_2 is handled in [/Pages/LatestPage.cshtml](https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/LatestPage.cshtml)

```csharp
﻿@page "/latest_{Page}"
```

[/Pages/Album.cshtml](https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/Album.cshtml) and [/Pages/AlbumPage.cshtml](https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/AlbumPage.cshtml) does the same for https://diffusion.works/albums/league-of-legends and https://diffusion.works/albums/league-of-legends_2

**Reply by mythz** [reply]
If it helps with code reuse, here's the code we use for shared partial we use for pagination:

 - https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/Shared/LatestNav.cshtml
 - https://github.com/NetCoreApps/VueDiffusion/blob/main/MyApp/Pages/Shared/AlbumNav.cshtml
